tool/tsh/proxy_test.go

```diff
@@ -176,6 +192,38 @@ func testJumpHostSSHAccess(t *testing.T, s *suite) {
 	})
 }
 
+// TestSSHLeafFromRoot tests that a user can ssh into a leaf node while logged in to
+// the root cluster without the --cluster flag if LoadAllHostCAs is true.
+func TestSSHLeafFromRoot(t *testing.T) {
```

Comment:
This test is failing for me locally (and I'm pretty sure that is also failing in CI). I don't think we can use `ssh` her. This test tries to connect to `localnode`, but this name won't resolve to localhost.
Also shouldn't we use the same code as `tsh` in our tests as changes were mainly there?

Reply:
I think the test failure was due to not setting the ping response in the right places. The other SSH tests in the package work fine with `localnode`, and the `ssh` approach covers the changes.